File: tests/test-terraria/src/main/java/pize/tests/terraria/indev/Tile.java
package pize.tests.terraria.indev;

import pize.graphics.texture.Texture;

public class Tile extends ITile{

    private TileTextures textures;

    public Tile(String id, Texture texture, TextureType textureType){
        super(id);

        textures = TileTextures.generate(texture, textureType);
    }

}

File: tests/test-terraria/src/main/kotlin/pize/tests/terraria/indev/Tile.kt
package pize.tests.terraria.indev

import pize.graphics.texture.Texture

class Tile(id: String?, texture: Texture?, textureType: TextureType) : ITile(id) {
    private val textures: TileTextures

    init {
        textures = TileTextures.Companion.generate(texture, textureType)
    }
}
